src/librustc_metadata/encoder.rs

```diff
@@ -266,15 +297,28 @@ impl<I, T> EncodeContentsForLazy<[T]> for I
     }
 }
 
+// Shorthand for `$self.$tables.$table.set($key, $self.lazy($value))`, which would
+// normally need extra variables to avoid errors about multiple mutable borrows.
```

Comment:
Isn't there a way to do this without macros? I imagine this to be really opaque to people trying to read the code.

Reply:
As the comment says, you'd need temporary variables (because you can't call `self.lazy(...)` inside the `set` call).
*However*, some of the uses of this macro only have only entry left by the last commit, so those could be nicer without the macro.

Maybe the answer is wrapping everything in helper methods.

Reply:
A regular method sounds preferable to me.